tests/equilibria/test_run.py

```diff
@@ -37,3 +47,101 @@ def test_Snapshot_copy(self):
                 assert cl != getattr(snapshot, sn.name)
             else:
                 assert cl == getattr(snapshot, sn.name)
+
+
+class TestPulsedCoilSetDesign:
+    def setup_method(self):
+        self.params = PulsedCoilsetDesignFrame.from_dict(
+            {
+                "A": {"value": 1, "unit": ""},
+                "B_premag_stray_max": {"value": 2, "unit": ""},
+                "C_Ejima": {"value": 3, "unit": ""},
+                "I_p": {"value": 4, "unit": ""},
+                "l_i": {"value": 5, "unit": ""},
+                "R_0": {"value": 6, "unit": ""},
+                "tau_flattop": {"value": 7, "unit": ""},
+                "tk_sol_ib": {"value": 8, "unit": ""},
+                "v_burn": {"value": 9, "unit": ""},
+            }
+        )
+
+    class MyPulsedCoilset(PulsedCoilsetDesign):  # noqa: D106
+        def optimise(self):
+            return self.coilset
+
+    def test_breakdown_settings(self):
+        mypcs = self.MyPulsedCoilset(self.params, *[None] * 6)
+        assert isinstance(mypcs._bd_settings, BreakdownCOPSettings)
+        mypcs._bd_settings.n_B_stray_points = 9
+        assert mypcs._bd_settings.n_B_stray_points == 9
+
+        mypcs = self.MyPulsedCoilset(
+            self.params, *[None] * 6, breakdown_settings={"n_B_stray_points": 10}
+        )
+        assert isinstance(mypcs._bd_settings, BreakdownCOPSettings)
+        assert mypcs._bd_settings.n_B_stray_points == 10
+
+        with pytest.raises(TypeError):
+            mypcs = self.MyPulsedCoilset(
+                self.params, *[None] * 6, breakdown_settings={"gamma": 1e-5}
+            )
+
+    def test_equilibrium_settings(self):
+        mypcs = self.MyPulsedCoilset(self.params, *[None] * 6)
+        assert isinstance(mypcs._eq_settings, EQSettings)
+        mypcs._eq_settings.gamma = 9
+        assert mypcs._eq_settings.gamma == 9
+
+        mypcs = self.MyPulsedCoilset(
+            self.params, *[None] * 6, equilibrium_settings={"gamma": 1e-5}
+        )
+        assert isinstance(mypcs._eq_settings, EQSettings)
+        assert mypcs._eq_settings.gamma == 1e-5
+
+        with pytest.raises(TypeError):
+            mypcs = self.MyPulsedCoilset(
+                self.params, *[None] * 6, equilibrium_settings={"n_B_stray_points": 10}
+            )
+
+    @mock.patch("bluemira.equilibria.run.Snapshot", return_value="SNAP")
+    def test_take_snapshot(self, snapshot, caplog):
+        mypcs = self.MyPulsedCoilset(self.params, *[None] * 6)
+        mypcs.take_snapshot("test", "eq", "coilset", "problem", "profiles")
+        snapshot.assert_called_with("eq", "coilset", "problem", "profiles", limiter=None)
+
+        assert mypcs.snapshots["test"] == "SNAP"
+        # Overwrite snapshot
+        mypcs.take_snapshot("test", "eq", "coilset", "problem", "profiles")
+        assert len(caplog.messages) == 1
+
+    # def test_run_premagnetisation(self):
+    # def test_run_reference_equilibrium(self):
```

Comment:
I'm guessing these are TODOs? Have we got an issue open to implement these?

Reply:
#1921